### ISSUE TITLE
Using CCE for image classification as loss

### ISSUE THREAD

**Issue by MichaelMMeskhi** [apple/ml-cross-entropy#22]
Out of curiosity, would CCE be easily interchanged with CE for instance in image classification problems?

**Comment by erikwijmans** [comment]
Should be. We developed CCE in the context of LLMs, but the only requirement of CCE is that there is a final affine transformation that transforms from the embedding space to logits over the label (vocab) space. So as long as your loss computation is something like 

```python
logits = self.classifier(embeddings)
loss = torch.nn.functional.cross_entropy(logits, labels)
```
where `self.classifier = nn.Linear(...)`

you can instead do
```python
from cut_cross_entropy import linear_cross_entropy
loss = linear_cross_entropy(embeddings, self.classifier.weight, labels, bias=self.classifier.bias)
```

**Comment by nleroy917** [comment]
Hello @erikwijmans! I might hijack this.

Thank you for this code, I'm very excited about the prospect of CCE; I have a data modality with extraordinarily large vocab sizes.

I tried implementing what you suggested above, and while it works (it runs), the loss appears to stay constant (right at 2). Im curious if you have any off-rip thoughts about this?

Here is exactly what I am running:
```python
def compute_cce_loss(self, embeddings: torch.Tensor, labels: torch.Tensor) -> torch.Tensor:
    """
    Compute the loss for the batch using the LigerFusedCrossEntropyLoss.

    Inspired by: https://github.com/apple/ml-cross-entropy/issues/22

    :param embeddings: The embeddings from the model of shape (batch_size, seq_length, d_model).
    :param labels: The labels tensor of shape (batch_size, seq_length).

    :return: The computed loss.
    """
    labels = labels.view(-1)
    embeddings = embeddings.view(-1, self.r2v_model.d_model)

    loss = linear_cross_entropy(
        embeddings, self.linear.weight, labels, bias=self.linear.bias, ignore_index=-100
    )

    return loss
```

<img width="560" alt="Image" src="https://github.com/user-attachments/assets/5bc876d6-02f8-4ae4-91bf-54e93b479bd9" />

FWIW, I am running with `lightning` and using `wandb` to track experiments. This is a single training step:
```python
def training_step(
    self, batch: Tuple[torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor], batch_idx: int
) -> torch.Tensor:
    """
    Perform a training step.

    :param batch: The batch containing (tokens, masked_tokens, labels, attention_mask).
    :param batch_idx: The batch index.
    :return: The computed loss.
    """
    _, masked_tokens, labels, attention_mask = batch
    masked_token_positions = labels != -100
    # forward pass
    embeddings = self.forward(
        masked_tokens, mask=attention_mask, masked_token_positions=masked_token_positions
    )
    # compute loss
    # loss = self.compute_loss(embeddings, labels)
    # loss = self.compute_fused_loss(embeddings, labels)
    loss = self.compute_cce_loss(embeddings, labels)

    # log loss
    self.log("train_loss", loss, on_step=True, on_epoch=True, prog_bar=True, logger=True)

    return loss
```
